File: Printpress.UI/Printpress_Angular/src/app/features/reports/models/inventory-services-usage-report.dto.ts
export interface InventoryItemUsageRowDto {
  itemCategory: string;
  itemName: string;
  packsPerCarton: number | null;
  unitsPerPack: number | null;
  cartonsIn: number;
  unitsIn: number;
  cartonsOut: number;
  unitsOut: number;
  periodNetCartons: number;
  periodNetUnits: number;
  currentStockCartons: number;
  currentStockUnits: number;
  expectedProductionWastePercent: number;
}

export interface ServiceUsageRowDto {
  serviceName: string;
  orderCount: number;
  itemCount: number;
  paperUsed: number;
}

export interface InventoryServicesUsageReportDto {
  inventoryItems: InventoryItemUsageRowDto[];
  totalCartonsIn: number;
  totalUnitsIn: number;
  totalCartonsOut: number;
  totalUnitsOut: number;
  totalPeriodNetCartons: number;
  totalPeriodNetUnits: number;
  totalCurrentStockCartons: number;
  totalCurrentStockUnits: number;
  services: ServiceUsageRowDto[];
  totalOrders: number;
  totalItems: number;
  totalPaperUsed: number;
}

export interface ServiceCategoryFilterDto {
  id: string;
  name: string;
}

File: Printpress.Application/Reports/DTO/InventoryServicesUsageReportDto.cs
namespace Printpress.Application;

public class InventoryServicesUsageReportDto
{
    public List<InventoryItemUsageRowDto> InventoryItems { get; set; }
    public int TotalCartonsIn { get; set; }
    public int TotalUnitsIn { get; set; }
    public int TotalCartonsOut { get; set; }
    public int TotalUnitsOut { get; set; }
    public int TotalPeriodNetCartons { get; set; }
    public int TotalPeriodNetUnits { get; set; }
    public int TotalCurrentStockCartons { get; set; }
    public int TotalCurrentStockUnits { get; set; }

    public List<ServiceUsageRowDto> Services { get; set; }
    public int TotalOrders { get; set; }
    public int TotalItems { get; set; }
    public decimal TotalPaperUsed { get; set; }
}
